### PR TITLE
[BEAM-6777] Add HealthDaemon and tests

### PR DESCRIPTION
**Pull request by rohdesamuel**
Adds a HealthDaemon and tests. The HealthDaemon will be used to communicate to the Dataflow service that the container is healthy.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by rohdesamuel** [comment]
R: @pabloem

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Ok i need to look at the design doc. I'll review this tomorrow.

File: sdks/python/apache_beam/runners/worker/health_daemon.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  SDK Harness is alive. If this SDK Harness does not communicate to the health
+  server after a configured amount of time, the health server will restart the
+  container.
+
+  Expected Usage:
+    # The HealthDaemon is expected to spin forever, start it on a separate
+    # thread.
+    health_thread = threading.Thread(target=HealthDaemon(8080).start)
+
+    # Automatically kill the thread when the program exists.
```
[the remark this answers is not in the document]
Comment (rohdesamuel) [reply sdks/python/apache_beam/runners/worker/health_daemon.py]: Done

File: sdks/python/apache_beam/runners/worker/health_daemon.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      health_server(http.client.HTTPConnection): Connection to the health
+        server.
+
+    Returns:
+      True if the health ping succeeded, false otherwise.
+    """
+
+    success = False
+    try:
+      health_server.request('PUT', '/sdk')
```
[the remark this answers is not in the document]
Comment (rohdesamuel) [reply sdks/python/apache_beam/runners/worker/health_daemon.py]: Done

**Comment by pabloem** [comment]
This looks good. Can you please squash the commits into one?

**Comment by aaltay** [comment]
Is the expectation that this will ping an endpoint hosted by dataflow service or the runner harness?

**Comment by rohdesamuel** [comment]
> Is the expectation that this will ping an endpoint hosted by dataflow service or the runner harness?

Neither, this will ping the existing Healthchecker sibling container that already sends health checks to Dataflow. The Healthchecker container will then communicate the health to the Dataflow service.

**Comment by rohdesamuel** [comment]
Run Python PreCommit

**Pull request closed without merging**